File: 289. Game of Life/gameOfLife.java
class gameOfLife {
    public void solution(int[][] board) {
    	if (board.length == 0) {
    		return;
    	}
    	int width = board.length;
    	int length = board[0].length;
    	int[][] directions = {{-1, -1}, {-1, 0}, {-1, 1}, {0, -1}, {0, 1}, {1, -1}, {1, 0}, {1, 1}};
    	for (int i = 0; i < width; i++) {
    		for (int j = 0; j < length; j++) {
    			int count = 0;
    			for (int k = 0; k < 8; k++) {
    				int x = i + directions[k][0];
    				int y = j + directions[k][1];
    				if (x < 0 || x >= width || y < 0 || y >= length) {
    					continue;
    				}
    				count = count + (board[x][y] & 1);
    			}
    			if ( (board[i][j] & 1) > 0) {
    				if (count >= 2 && count <= 3) {
    					board[i][j] = 0b11;
    				}
    			}
    			else if (count == 3) {
    				board[i][j] = 0b10;
    			}
    		}
    	}
    	for (int i = 0; i < width; i++) {
    		for (int j = 0; j < length; j++) {
    			board[i][j] >>= 1;
    		}
    	}

    	printArry(board);
    }

    public void printArry(int[][] arr) {
    	for (int i = 0; i < arr.length; i++) {
    		for (int j = 0; j < arr[0].length; j++) {
    			System.out.print(arr[i][j] + " ");
    		}
    		System.out.println(" ");
    	}
    }

    public static void main(String[] argv) {
    	int[][] arr = {{0,1,0},{0,0,1},{1,1,1},{0,0,0}};
    	gameOfLife obj = new gameOfLife();
    	obj.printArry(arr);
    	obj.solution(arr);
    }
}

File: 289. Game of Life/gameOfLife.py
class gameOfLife:
    def solution(self, board):
        width = len(board)
        length = len(board[0])
        if not width:
            return
        directions = [[-1, -1], [-1, 1], [-1, 0], [0, -1], [0, 1], [1, -1], [1, 0], [1, 1]]
        for i in range(0, width):
            for j in range(0, length):
                count = 0
                for k in range(0, 8):
                    x = i + directions[k][0]
                    y = j + directions[k][1]
                    if x < 0 or x >= width or y < 0 or y >= length:
                        continue
                    count = count + (board[x][y] & 1)
                if board[i][j] & 1:
                    if count >= 2 and count <= 3:
                        board[i][j] = 0b11
                elif count == 3:
                    board[i][j] = 0b10
        for i in range(0, width):
            for j in range(0, length):
                board[i][j] >>= 1
            print(board[i])

obj = gameOfLife()
arr = [[0,1,0],[0,0,1],[1,1,1],[0,0,0]]
obj.solution(arr)